### PR TITLE
[Bug] [lang] Fix tuple assignment behavior in Taichi-scope

### LINKED ISSUE #1237: [Lang] Tuple assignment bug
**Describe the bug**
Tuple assignments in Taichi don't have the same behavior as those in Python.

**To Reproduce**
```python
import taichi as ti
ti.init(print_preprocessed=True)

@ti.kernel
def fibonacci_taichi():
    a, b = 0, 1
    a, b = b, a + b
    print(a, b)

def fibonacci_python():
    a, b = 0, 1
    a, b = b, a + b
    print(a, b)

fibonacci_taichi()
fibonacci_python()
```

Note that Taichi prints `2 1` while Python prints `1 1`.

In preprocessing, we should create a copy of the variables instead of using a reference (Expr):
```
Before preprocessing:
@ti.kernel
def fibonacci_taichi():
    a, b = 0, 1
    a, b = b, a + b
    print(a, b)

After preprocessing:
def fibonacci_taichi():
  if 1:
    __tmp_tuple = 0, 1
    a = ti.expr_init(__tmp_tuple[0])
    b = ti.expr_init(__tmp_tuple[1])
    del __tmp_tuple
  if 1:
    __tmp_tuple = b, a + b #### Maybe this should be ti.expr_init(a, a + b)?
    a.assign(__tmp_tuple[0])
    b.assign(__tmp_tuple[1])
    del __tmp_tuple
  ti.ti_print(a, b)
```

`transform.py` is probably the file to modify.

### PR DESCRIPTION
**Pull request by archibate**
Related issue = close #1237

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----
Just a very tiny modification.

### CODE REVIEW

**Comment by archibate** [comment]
Why my PR always no being responsed except bots? I think I made a good PR with far-more-complete test.

**Comment by archibate** [comment]
Btw, is it possible to unpack a vector? e.g.:
```
v = ti.Vector([2, 3, 4])
x, y, z = v
```

File: python/taichi/lang/impl.py [hunk 1]
```diff
@@ -28,6 +28,21 @@ def expr_init(rhs):
             return Expr(taichi_lang_core.expr_var(Expr(rhs).ptr))
 
 
+@taichi_scope
+def expr_init_list(xs, expected):
+    if not isinstance(xs, (list, tuple)):
```
Comment (k-ye) [inline python/taichi/lang/impl.py:33]: Is it possible to support `ti.Vector`?

Comment (archibate) [reply python/taichi/lang/impl.py:33]: Yes, as I mentioned above, let's fix this asap and open another issue on it.

**Comment by archibate** [comment]
@yuanming-hu Given that you are more occupied in GAMES 201, I'm merging this since it's trivial and got @k-ye's approvement.

**Pull request merged**